lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -124,6 +149,13 @@ class AwsCompileFunctions {
 
     _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Outputs,
       newOutputObject);
+
+    const newVersionOutput = this.cfOutputLatestVersionTemplate();
+    newVersionOutput.Value = { Ref: versionLogicalId };
+
+    _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Outputs, {
+      [`${functionLogicalId}QualifiedArn`]: newVersionOutput,
```

Comment:
For this one as well, could move the logical ID generation into [the `naming.js` file](https://github.com/serverless/serverless/blob/master/lib/plugins/aws/lib/naming.js)?

And I'd go with: `FunctionNameQualifiedArn` for the same reasons above